sklearn/neighbors/_binary_tree.pxi.tp

```diff
@@ -1143,10 +1143,10 @@ cdef class BinaryTree{{name_suffix}}:
 
         # flatten X, and save original shape information
         np_Xarr = X.reshape((-1, self.data.shape[1]))
-        cdef const {{INPUT_DTYPE_t}}[:, ::1] Xarr = np_Xarr
+        cdef {{INPUT_DTYPE_t}}[:, ::1] Xarr = np_Xarr
```

Comment:
This one is required (see [this error message](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=58158&view=logs&j=dde5042c-7464-5d47-9507-31bdd2ee0a3a&t=4bd2dad8-62b3-5bf9-08a5-a9880c530c94&l=4709)).
```suggestion
        cdef const {{INPUT_DTYPE_t}}[:, ::1] Xarr = np_Xarr
```